bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/servlets/FormMetaDataDataSourceServlet.java

```diff
@@ -131,57 +135,49 @@ protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHtt
         if (config != null) {
             FormMetaDataType type = FormMetaDataType.fromString(getParameter(config, TYPE, request, null));
             String dataModel = getParameter(config, DATA_MODEL, request, "");
-            String fieldType = getParameter(config, FIELD_TYPE, request, "");
             actionTypeDataSource = new SimpleDataSource(getDataSourceResources(
-                request.getResourceResolver(), type, dataModel, fieldType).iterator());
+                request, type, dataModel).iterator());
         }
         request.setAttribute(DataSource.class.getName(), actionTypeDataSource);
     }
 
-    private List<Resource> getDataSourceResources(ResourceResolver resourceResolver, FormMetaDataType type, String dataModel,
-        String fieldType) {
+    private List<Resource> getDataSourceResources(SlingHttpServletRequest request, FormMetaDataType type, String dataModel) {
```

Comment:
Can we use HttpServletRequest here ? Also, let's pass the resource resolver explicitly as argument.

Reply:
done